torch/csrc/autograd/VariableTypeManual.cpp

```diff
@@ -111,6 +112,15 @@ int64_t _version(const Tensor & self) {
   return as_variable_ref(self).current_version();
 }
 
+Tensor& requires_grad_(Tensor& self, bool _requires_grad) {
+  if (!self.is_leaf() && !_requires_grad) {
+    throw std::runtime_error(
+      autograd::utils::requires_grad_leaf_error(requires_grad)
```

Comment:
This is a bug.  `requires_grad` is a function, `_requires_grad` is the boolean.

Reply:
@pbelevich

Reply:
Fixed in https://github.com/pytorch/pytorch/commit/1e3e1f5bf9745d3c82dc0834064e443d6d142785